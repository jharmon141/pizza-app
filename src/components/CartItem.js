import React from 'react';
import { Col, Row } from 'react-bootstrap';
import PropTypes from 'prop-types';

let quantityNums = [];

for (let i = 1; i < 11; i++) {
   quantityNums.push(i);
};

export default class CartItem extends React.Component {

   constructor(props) {
      super(props);

      this.state = {
         quantity: 0
      };

      this.updatePizza = this.updatePizza.bind(this);
   }

   static propTypes = {
      handleUpdatePizza: PropTypes.func.isRequired,
      removePizza: PropTypes.func.isRequired,
      pizza: PropTypes.shape({
         id: PropTypes.number.isRequired,
         size: PropTypes.string.isRequired,
         basePrice: PropTypes.number.isRequired,
         toppings: PropTypes.arrayOf(PropTypes.shape({
            __typename: PropTypes.string.isRequired,
            name: PropTypes.string.isRequired,
            price: PropTypes.number.isRequired
         }).isRequired).isRequired
      }).isRequired   
   };

   updatePizza(event) {
      let pizza = this.props.pizza;
      pizza.quantity = Number(event.target.value);
      this.props.handleUpdatePizza(pizza);
      this.setState({
         quantity: event.target.value
      });
   };

   componentWillMount() {
      this.setState({
         quantity: this.props.pizza.quantity
      });
   };

   render() {
      return(
         <div>
            <Row>
               <Col xs={2} xsOffset={1}>
                  <h4 >{this.props.pizza.size}</h4>
               </Col>
               <Col  className="cartItem" xs={3}>
                  <span>
                     {this.props.pizza.toppings.map((topping, i) => {
                        if (this.props.pizza.toppings[i+1]) {
                           return <span key={topping.name}>{topping.name}, </span>
                        }
                        else {
                           return <span key={topping.name}>{topping.name}</span>
                        }
                     })}
                  </span>
               </Col>
               <Col xs={2}> 
                  <select 
                     className="quantitySelect"
                     onChange={this.updatePizza} 
                     value={this.state.quantity}>
                     {quantityNums.map((num) => {
                        return <option
                           key={num}
                           value={num}>
                           {num}
                        </option>
                     })}
                  </select>
               </Col>
               <Col xs={1}>
                  <h4>${(this.props.pizza.basePrice * this.props.pizza.quantity).toFixed(2)}</h4>
               </Col>
               <Col xs={1}>
                  <h4 className="delete" onClick={() => this.props.removePizza(this.props.pizza)}>X</h4>
               </Col>
            </Row>
         </div>
      );
   };
};
