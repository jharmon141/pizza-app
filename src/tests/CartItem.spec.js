import React from 'react';
import CartItem from '../components/CartItem';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('Cart Component', () => {
   it('renders without crashing', () => {
      let pizza = {
         id: 1,
         size: 'small',
         basePrice: 9.00,
         toppings: []
      };

      const tree = renderer.create(
         <MemoryRouter> 
            <CartItem 
               pizza={pizza}
               removePizza={() => {}}
               handleUpdatePizza={() => {}}
            />
         </MemoryRouter>
      ).toJSON();
      expect(tree).toMatchSnapshot();
   });
});
