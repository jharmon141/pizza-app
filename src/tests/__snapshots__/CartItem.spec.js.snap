// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Component renders without crashing 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xs-2 col-xs-offset-1"
    >
      <h4>
        small
      </h4>
    </div>
    <div
      className="cartItem col-xs-3"
    >
      <span />
    </div>
    <div
      className="col-xs-2"
    >
      <select
        className="quantitySelect"
        onChange={[Function]}
        value={undefined}
      >
        <option
          value={1}
        >
          1
        </option>
        <option
          value={2}
        >
          2
        </option>
        <option
          value={3}
        >
          3
        </option>
        <option
          value={4}
        >
          4
        </option>
        <option
          value={5}
        >
          5
        </option>
        <option
          value={6}
        >
          6
        </option>
        <option
          value={7}
        >
          7
        </option>
        <option
          value={8}
        >
          8
        </option>
        <option
          value={9}
        >
          9
        </option>
        <option
          value={10}
        >
          10
        </option>
      </select>
    </div>
    <div
      className="col-xs-1"
    >
      <h4>
        $
        NaN
      </h4>
    </div>
    <div
      className="col-xs-1"
    >
      <h4
        className="delete"
        onClick={[Function]}
      >
        X
      </h4>
    </div>
  </div>
</div>
`;
